Text on session resumption (#3566)
quicwg/base-drafts: draft-ietf-quic-tls.md

of type PROTOCOL_VIOLATION.

## Session Resumption {#resumption}

QUIC can use the session resumption feature of TLS 1.3. It does this by
carrying NewSessionTicket messages in CRYPTO frames after the handshake is
complete. Session resumption is the basis of 0-RTT, but can be used without
also enabling 0-RTT.

> Endpoints that use sesion resumption might need to remember some information

Reviewer: ```suggestion
Endpoints that use session resumption might need to remember some information
```

about the current connection when creating a resumed connection. TLS requires
that some information be retained; see Section 4.6.1 of {{!TLS13}}. QUIC itself
does not depend on any state being retained when resuming a connection, unless
0-RTT is also used; see {{enable-0rtt}} and Section 7.3.1 of
{{QUIC-TRANSPORT}}. Application protocols could depend on state that is
retained between resumed connections.

Clients can store any state required for resumption along with the session
ticket. Servers can use the session ticket to help carry state.

Session resumption allows servers to link activity on the original connection
with the resumed connection, which might be a privacy issue for clients.
Clients can choose not to enable resumption to avoid creating this correlation.

> Client SHOULD NOT reuse tickets as that allows entities other than the server

Reviewer: ```suggestion
Client SHOULD NOT use a single ticket more than once as that allows entities other than the server
```

to correlate connection; see Section C.4 of {{!TLS13}}.

> 

Reviewer 2: Looks good. Simple and to the point. My only reservation is about the security section. Session resumption allows tracking by the server, and this is arguably a security issue. The text here properly describes the concern and the remediation, but I wonder whether should there be a mention of these potential privacy issues in the security section.

> to correlate connection; see Section C.4 of {{!TLS13}}.

Reviewer 3: ```suggestion
to correlate connections; see Section C.4 of {{!TLS13}}.
```
